src/client/containers/PullRequestSummary/ReviewSection/index.js

```diff
@@ -141,32 +139,29 @@ class ReviewSection extends Component {
 
   props: ReviewersSectionProps
 
-  reviewerSelectionChanged: (string, Array<UserType>) => void
-
-  reviewerSelectionChanged(id: string, reviewers: Array<UserType>) : void {
+  reviewerSelectionChanged = (id: string, reviewers: Array<UserType>): void => {
     this.setState({
       selection: reviewers,
     })
   }
 
-  removeReviewer: UserType => void
-
-  removeReviewer(reviewer: UserType) : void {
+  removeReviewer = (reviewer: UserType): void => {
     this.setState(
       {
         removed: this.state.removed.concat(reviewer),
       }
     )
   }
 
-  saveReviewerChanges: () => void
+  handleSaveReviewers = (): void => {
+    const removedReviewers = this.state.removed.map(r => ({ id: r.id }))
+    const addedReviewers = this.state.selection.map(r => ({ id: r.id }))
 
-  saveReviewerChanges() : void {
-    const toRemove = this.state.removed.map(r => ({ id: r.id }))
-    const toAdd = this.state.selection.map(r => ({ id: r.id }))
-
-    if (toRemove.length > 0 || toAdd.length > 0) {
-      this.props.changeReviewers(this.props.id, toAdd, toRemove)
+    if (addedReviewers.length) {
+      this.props.dispatch(addPullRequestReviewers(this.props.id, addedReviewers))
+    }
+    if (removedReviewers.length) {
+      this.props.dispatch(removePullRequestReviewers(this.props.id, removedReviewers))
```

Comment:
The order of addPullRequestReviewers and removePullRequestReviewers should be inverted.
If we remove a reviewer and add him again he would be removed in this ordering; we want him to stay in this case, so either we should filter removedReviewers or invoke in the order of removePullRequestReviewers then addPullRequestReviewers